File: 260-single-number-iii/single-number-iii.java
class Solution {
    public int[] singleNumber(int[] nums) {
        int ans[] = new int[2];
        int xor = 0;
        for(int i:nums){
            xor^=i;
        }
        int first = 0;
        int mask = (xor&(-1*xor));
        for(int i:nums){
            if((mask&i)==mask){
                first^=i;
            }
        }
        ans[0]=first;
        ans[1]=(first^xor);
        return ans;
    }
}

File: 260-single-number-iii/single-number-iii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        xor = 0
        for i in nums:
            xor = xor^i
        mask = xor&(-1*xor)
        first = 0
        for i in nums:
            if((mask&i)==mask):
                first = first^i
        second = first^xor
        return [first,second]
